src/components/button/styles.ts

```diff
@@ -0,0 +1,66 @@
+import styled, { css } from 'styled-components';
+import {
+  colors,
+  colorsRgba,
+  mediaQueries,
+  rgbaMap,
+} from '../../styles/constants';
+
+interface ButtonProps {
```

Comment:
you could reuse this type. First, declare it in the button.tsx like this:
`export type ButtonType =  'submit' | 'button' | 'reset';`
then just reuse it here and at the same time in prop type of the component button.tsx:

`
interface ButtonProps  {
 ...,
 type: ButtonType,
 ...
}
`

If your plan to have type 'submit' but styles of 'button' then this approach is OK. I would only add this to props so the  reader of the code can see what's happening

Reply:
Good point, thnkx